rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -196,33 +198,35 @@ func (r *Layer2Relayer) initializeGenesis() error {
 			return fmt.Errorf("failed to update genesis chunk proving status: %v", err)
 		}
 
-		batchMeta := &types.BatchMeta{
+		batch := &encoding.Batch{
```

Comment:
Better to explicitly initialize all fields.

```suggestion
		batch := &encoding.Batch{
			Index: 0,
			TotalL1MessagePoppedBefore: 0,
			ParentBatchHash: common.Hash{},
```

Reply:
Nice catch, this commit genesis batch path is important, also changed another part (in unit test).